Source/WebCore/page/scrolling/ScrollSnapOffsetsInfo.cpp

```diff
@@ -463,7 +482,21 @@ FloatScrollSnapOffsetsInfo LayoutScrollSnapOffsetsInfo::convertUnits(float devic
 template <> template <>
 std::pair<LayoutUnit, std::optional<unsigned>> LayoutScrollSnapOffsetsInfo::closestSnapOffset(ScrollEventAxis axis, const LayoutSize& viewportSize, LayoutPoint scrollDestinationOffset, float velocity, std::optional<LayoutUnit> originalPositionForDirectionalSnapping) const
 {
-    return closestSnapOffsetWithInfoAndAxis(*this, axis, viewportSize, scrollDestinationOffset, velocity, originalPositionForDirectionalSnapping);
+    auto horizontal = closestSnapOffsetWithInfoAndAxis(*this, ScrollEventAxis::Horizontal, viewportSize, scrollDestinationOffset, velocity, originalPositionForDirectionalSnapping);
+    auto vertical = closestSnapOffsetWithInfoAndAxis(*this, ScrollEventAxis::Vertical, viewportSize, scrollDestinationOffset, velocity, originalPositionForDirectionalSnapping);
+    const auto& snapOffsetsHorizontal = offsetsForAxis(ScrollEventAxis::Horizontal);
+    const auto& snapOffsetsVertical= offsetsForAxis(ScrollEventAxis::Vertical);
+
+    if (horizontal.second && vertical.second && !offsetHasVisisbleSnapArea(*this, snapOffsetsHorizontal[*horizontal.second], snapOffsetsVertical[*vertical.second], ScrollEventAxis::Horizontal, viewportSize) && !offsetHasVisisbleSnapArea(*this, snapOffsetsVertical[*vertical.second], snapOffsetsHorizontal[*horizontal.second], ScrollEventAxis::Vertical, viewportSize)) {
+        
+        auto horizontalSnapArea = snapAreas[snapOffsetsHorizontal[*horizontal.second].snapAreaIndices[compatibleSnapArea(*this, snapOffsetsHorizontal[*horizontal.second], ScrollEventAxis::Horizontal, viewportSize, scrollDestinationOffset)]];
+        auto verticalSnapArea = snapAreas[snapOffsetsVertical[*vertical.second].snapAreaIndices[compatibleSnapArea(*this, snapOffsetsVertical[*vertical.second], ScrollEventAxis::Vertical, viewportSize, scrollDestinationOffset)]];
+        
+        auto closerSnapArea = (horizontalSnapArea.x() - scrollDestinationOffset.x())* (horizontalSnapArea.x() - scrollDestinationOffset.x()) + (horizontalSnapArea.y() - scrollDestinationOffset.y())* (horizontalSnapArea.y() - scrollDestinationOffset.y()) > (verticalSnapArea.x() - scrollDestinationOffset.x())*(verticalSnapArea.x() - scrollDestinationOffset.x()) + (verticalSnapArea.y() - scrollDestinationOffset.y())*(verticalSnapArea.y() - scrollDestinationOffset.y()) ? verticalSnapArea : horizontalSnapArea;
```

Comment:
Nit - space before `*` in `scrollDestinationOffset.x())*`

Reply:
Fixed.